### PR TITLE
[BOB-16] Add parser decorator/proxy to create pkb entities using ipkb creator

### PR DESCRIPTION
**Pull request by bciereszynski**

### CODE REVIEW

File: src/SpaFronted/Parsobober.DesignExtractor/ProgramContextExtractor.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
+{
+    internal class ProgramContextExtractor(IProgramContextCreator creator) : SimpleExtractor
+    {
+        private readonly IProgramContextCreator creator = creator;
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/ProgramContextExtractor.cs:8]: you have 2 same private props `creator` one from primary constructors and second one you wrote

File: src/SpaFronted/Parsobober.DesignExtractor/FollowsExtractor.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
+{
+    internal class FollowsExtractor(IFollowsCreator creator) : SimpleExtractor
+    {
+        private readonly IFollowsCreator creator = creator;
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/FollowsExtractor.cs:8]: same

File: src/SpaFronted/Parsobober.DesignExtractor/FollowsExtractor.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/FollowsExtractor.cs:4]: change all namespaces to file scoped pls

File: src/SpaFronted/Parsobober.DesignExtractor/ModifiesExtractor.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/ModifiesExtractor.cs:4]: 💀

File: src/SpaFronted/Parsobober.DesignExtractor/ModifiesExtractor.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+{
+    internal class ModifiesExtractor(IModifiesCreator creator) : SimpleExtractor
+    {
+        private readonly IModifiesCreator creator = creator;
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/ModifiesExtractor.cs:8]: 💀

File: src/SpaFronted/Parsobober.DesignExtractor/ParentExtractor.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/ParentExtractor.cs:4]: 💀

File: src/SpaFronted/Parsobober.DesignExtractor/ParentExtractor.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+{
+    internal class ParentExtractor(IParentCreator creator) : SimpleExtractor
+    {
+        private readonly IParentCreator creator = creator;
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/ParentExtractor.cs:8]: 💀

File: src/SpaFronted/Parsobober.DesignExtractor/ProgramContextExtractor.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/ProgramContextExtractor.cs:4]: 💀

File: src/SpaFronted/Parsobober.DesignExtractor/SimpleExtractor.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Simple.Parser.Abstractions;
+
+namespace Parsobober.Simple.Parser.Extractor
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/SimpleExtractor.cs:4]: 💀

File: src/SpaFronted/Parsobober.DesignExtractor/UsesExtractor.cs [hunk 10]
```diff
@@ -0,0 +1,47 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
+{
+    internal class UsesExtractor(IUsesCreator creator) : SimpleExtractor
+    {
+        private readonly IUsesCreator creator = creator;
```
Comment (pzajaczkowski) [inline src/SpaFronted/Parsobober.DesignExtractor/UsesExtractor.cs:8]: 💀

**Review by pzajaczkowski — CHANGES_REQUESTED** [review]

File: src/SpaFronted/Parsobober.DesignExtractor/ModifiesExtractor.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        private Stack<TreeNode> stack = new Stack<TreeNode>();
+
+        public override void While(TreeNode result)
+        {
+            Stack<TreeNode> tempStack = new Stack<TreeNode>();
```
Comment (maksimowiczm) [inline src/SpaFronted/Parsobober.DesignExtractor/ModifiesExtractor.cs:13]: var

**Review by maksimowiczm — APPROVED** [review]

**Review by pzajaczkowski — APPROVED** [review]

**Pull request merged**